### PR TITLE
feat: rewrite headers of a redirect

### PR DESCRIPTION
**Pull request by snylonue**
replaces #26

### CODE REVIEW

File: lib/utils/redirect_interceptor.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    final ResponseInterceptorHandler handler,
+  ) {
+    if (headers == null &&
+        response.statusCode! >= 300 &&
+        response.statusCode! < 400) {
+      headers = response.headers;
+    } else if (headers != null &&
+        response.statusCode! >= 200 &&
+        response.statusCode! < 300) {
+      response.headers = headers!;
```
Comment (Yesterday17) [inline lib/utils/redirect_interceptor.dart]: 这里是不是要重新吧 headers 设置为 null？

Comment (snylonue) [reply lib/utils/redirect_interceptor.dart]: 如果同时有多个请求会不会有问题

Comment (Yesterday17) [reply lib/utils/redirect_interceptor.dart]: 确实，那可能要标记一下请求 id？

**Review by Yesterday17 — APPROVED** [review]
Merging this pull request as a temporary fix. We'll eventually move all http-related parts to rust.

**Pull request merged**